File: packages/renderer-vue/playground/measurements/FilterToTypeNode.ts
import { defineNode, NodeInterface } from "@baklavajs/core";
import { setType } from "@baklavajs/interface-types";
import { Measurement } from "../models/Measurement";
import { measurementListType, stringType } from "../interfaceTypes";
import { TextInputInterface } from "../../src";

export default defineNode({
    type: "FilterToTypeNode",
    title: "Filter To Type",
    inputs: {
        inputMeasurements: () => new NodeInterface<Measurement[]>("Measurements", []).use(setType, measurementListType),
        type: () => new TextInputInterface("measurement type", "").use(setType, stringType),
    },
    outputs: {
        outputMeasurements: () => new NodeInterface<Measurement[]>("Filtered Measurements", []).use(setType, measurementListType),
    },
    calculate(inputs) {
        const outputMeasurements: Measurement[] = [];
        if (Array.isArray(inputs.inputMeasurements)) {
            for (const measurement of inputs.inputMeasurements) {
                if (measurement && typeof measurement.type === 'string' && measurement.type == inputs.type) {
                    outputMeasurements.push(measurement);
                }
            }
        }

        return { outputMeasurements };
    },
});

File: BaklavaDependencyEngine/BaklavaDependencyEngine.Core/ExampleNodes/FilterToTypeNode.cs
using BaklavaDependencyEngine.Core.Models;

namespace BaklavaDependencyEngine.Core.ExampleNodes
{
    public class FilterToTypeNode : Node
    {
        public FilterToTypeNode() : base("FilterToTypeNode", "Filter To Type")
        {
        }

        protected override void InitializeInterfaces()
        {
            AddInput("inputMeasurements", new NodeInterface("Measurements", new List<Measurement>()));
            AddInput("type", new NodeInterface("Type", ""));
            AddOutput("outputMeasurements", new NodeInterface("Filtered Measurements", new List<Measurement>(), false));
        }

        public override async Task<Dictionary<string, object>> Calculate(
            Dictionary<string, object> inputs,
            CalculationContext context)
        {
            var outputMeasurements = new List<Measurement>();
            var typeFilter = inputs["type"]?.ToString() ?? "";

            if (inputs["inputMeasurements"] is List<Measurement> measurements)
            {
                foreach (var measurement in measurements)
                {
                    if (measurement != null && measurement.Type == typeFilter)
                    {
                        outputMeasurements.Add(measurement);
                    }
                }
            }

            return await Task.FromResult(new Dictionary<string, object>
            {
                ["outputMeasurements"] = outputMeasurements
            });
        }
    }
}